linode_api4/linode_client.py

```diff
@@ -51,26 +64,57 @@ def __init__(
                                   can be found in the API docs, but at time of writing
                                   are between 25 and 500.
         :type page_size: int
-        :param retry_rate_limit_interval: If given, 429 responses will be automatically
-                                         retried up to 5 times with the given interval,
-                                         in seconds, between attempts.
-        :type retry_rate_limit_interval: int
+        :param retry: Whether API requests should automatically be retries on known
+                      intermittent responses.
+        :type retry: bool
+        :param retry_rate_limit_interval: The amount of time to wait between HTTP request
+                                          retries.
+        :type retry_rate_limit_interval: float
+        :param retry_max: The number of request retries that should be attempted before
+                          raising an API error.
+        :type retry_max: int
+        :type retry_statuses: List of int
+        :param retry_statuses: Additional HTTP response statuses to retry on.
+                               By default, the client will retry on 408 and 429
+                               responses.
         """
         self.base_url = base_url
         self._add_user_agent = user_agent
         self.token = token
-        self.session = requests.Session()
         self.page_size = page_size
-        self.retry_rate_limit_interval = retry_rate_limit_interval
+
+        retry_forcelist = [408, 429]
```

Comment:
```suggestion
        retry_forcelist = [408, 429, 500, 502]
```
Can we also add 500 and 502?